CHANGELOG

```diff
@@ -1,6 +1,21 @@
 UNRELEASED:
-        * Add 'Path.Posix' and 'Path.Windows' modules for manipulating
-          Windows or Posix style paths independently of the current platform.
+	* (BREAKING CHANGE)
+		"addFileExtension" and "setFileExtension" now accept only "valid"
+		extension forms and "fileExtension" also returns the same form.  A
+		valid extension starts with a @.@ followed by one or more characters
+		not including @.@ followed by zero or more @.@s in trailing position.
+		You may have to replace invalid forms of extensions with valid ones.
+		Especially:
+		* "foo" was earlier equivalent to ".foo" but is now invalid
+		* ".foo.bar" is now invalid, instead set ".foo" and ".bar" separately
```

Comment:
Think about this law:

```
(fileExtension . fromJust . addFileExtension ext) file == ext
```
If we allow `.tar.gz` to be set then `fileExtension` will return `.gz` and this law will not hold. However, if we set one extension at a time then this law always holds.  But it is not strictly necessary that we obey this law, we have the option to allow extensions like`.tar.gz` and say that this is not a law.

Reply:
Sounds good to me.

Reply:
I am trying to interpret your response. Does that mean we should allow `.tar.gz`? In fact I am tending towards allowing it because majority of users will intuitively think that it should be allowed. And this law does not sound very important to me.